src/plrust.rs

```diff
@@ -90,8 +92,9 @@ pub(crate) fn compile_function(fn_oid: pg_sys::Oid) -> eyre::Result<(PathBuf, Ou
     let work_dir = gucs::work_dir();
     let pg_config = gucs::pg_config();
     let target_dir = work_dir.join("target");
+    let db_oid = unsafe { MyDatabaseId };
```

Comment:
When exactly does this global variable get initialized by Postgres? And does it conceivably get altered during runtime, or is it strictly write-once from our perspective? It would be ideal if we documented the safety conditions for this, as bindgen says this is a `static mut` and thus racy.

Reply:
It really only gets set once, unless some extensions do weird things but we shouldn't worry about those.
I ran against the latest postgres code:

```
desktop  % grep -ri "MyDatabaseId = " *
src/backend/utils/init/globals.c:Oid			MyDatabaseId = InvalidOid;
src/backend/utils/init/postinit.c:		MyDatabaseId = Template1DbOid;
src/backend/utils/init/postinit.c:		MyDatabaseId = dbform->oid;
src/backend/utils/init/postinit.c:		MyDatabaseId = dbform->oid;
```
the postinit.c ones are only called when a Postgres backend is initialized:

https://github.com/postgres/postgres/blob/master/src/backend/utils/init/postinit.c#L664

globals.c just initializes the value for the first time:
https://github.com/postgres/postgres/blob/master/src/backend/utils/init/globals.c#L89

Reply:
Ah. The global definition means it is always safe to read in Rust terms (i.e. is not uninitialized data). And yes, I think it is safe to assume we're not racing in practice.
```suggestion
    // SAFETY: Postgres globally sets this to `const InvalidOid`, so is always read-safe,
    // then writes it only during initialization, so we should not be racing anyone.
    let db_oid = unsafe { MyDatabaseId };
```